vscode/microsoft-kiota/src/commands/deleteWorkspaceItem/deleteWorkspaceItemCommand.ts

```diff
@@ -37,8 +37,10 @@ export class DeleteWorkspaceItemCommand extends Command {
       if (result) {
         const isSuccess = result.some(k => k.message.includes('removed successfully'));
         if (isSuccess) {
-          void vscode.window.showInformationMessage(vscode.l10n.t('{0} removed successfully.', workspaceTreeItem.label));
+          // add a delay to ensure the workspace tree is refreshed after the item is removed from the file
+          await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
I don't think we should introduce artificial delays in the code.
It doesn't guarantee the work is completed behind the scene (what if the system is under load and it takes longer than 1s?)
It slows down the experience for everyone (what if the work takes less than 1s on average)

This is a sign that a promise is not awaited somewhere in the chain, we should track this down instead.

Reply:
yes, fire and forget is dangerous, we can always review the approach if users complain about the extension being slow.